File: lib-util/src/main/java/code/util/AppGlobal.java
package code.util;

import android.app.Application;

import java.lang.reflect.InvocationTargetException;

public class AppGlobal {
    public static final String TAG = AppGlobal.class.getSimpleName();
    private static Application sApp;

    public static Application getApplication() {
        if (sApp == null) {
            try {
                sApp = (Application) Class.forName("android.app.ActivityThread")
                        .getMethod("currentApplication")
                        .invoke(null, (Object[]) null);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        return sApp;
    }
}

File: lib-util/src/main/java/code/util/AppGlobal.kt
package code.util

import android.app.Application

object AppGlobal {
    private val TAG = AppGlobal::class.java.simpleName
    private var sApp: Application? = null

    fun getApplication(): Application {
        if (sApp == null) {
            try {
                sApp = Class.forName("android.app.ActivityThread")
                    .getMethod("currentApplication")
                    .invoke(null) as Application
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        return sApp!!
    }

}